RadixWallet/Features/SplashFeature/Splash.swift

```diff
@@ -23,40 +32,43 @@ public struct Splash: Sendable, FeatureReducer {
 
 	public enum InternalAction: Sendable, Equatable {
 		case passcodeConfigResult(TaskResult<LocalAuthenticationConfig>)
-		case loadedProfile(Profile)
-		case accountRecoveryNeeded(TaskResult<Bool>)
+		case biometricsCheckResult(TaskResult<Bool>)
+		case checkBiometrics
+		case loadingDataCompleted
 	}
 
 	public enum DelegateAction: Sendable, Equatable {
 		case completed(Profile)
 	}
 
 	public struct Destination: DestinationReducer {
+		@CasePathable
 		public enum State: Sendable, Hashable {
 			case passcodeCheckFailed(AlertState<Action.PasscodeCheckFailedAlert>)
+			case errorAlert(AlertState<Action.ErrorAlert>)
 		}
 
+		@CasePathable
 		public enum Action: Sendable, Equatable {
 			case passcodeCheckFailed(PasscodeCheckFailedAlert)
+			case errorAlert(ErrorAlert)
```

Comment:
I find it a bit confusing that we have:
- `state.biometricsCheckFailed`: boolean used to show something or not on view
- `state.destination.passcodeCheckFailed`: a destination alert
- `state.destination.errorAlert`: a destination alert

We can probably merge the last two in one single destination (`errorAlert`), that depending on the situation that triggers it we populate `AlertState` with different title, message and buttons.

Reply:
I thought it would be clearer to keep them separate, but yes, I agree we can merge them.